test/core/compression/message_compress_test.cc

```diff
@@ -210,12 +210,10 @@ static void test_bad_decompression_data_missing_trailer(void) {
   grpc_core::ExecCtx exec_ctx;
   /* compress it */
   grpc_msg_compress(GRPC_MESSAGE_COMPRESS_GZIP, &input, &corrupted);
-  /* corrupt the output by smashing the CRC */
-  GPR_ASSERT(corrupted.count > 1);
-  GPR_ASSERT(GRPC_SLICE_LENGTH(corrupted.slices[1]) > 8);
+  GPR_ASSERT(GRPC_SLICE_LENGTH(corrupted.slices[corrupted.count - 1]) > 8);
+  /* Remove the footer by manipulating the slice length */
```

Comment:
Hmm it's definitely better, but what if there are 3 slices or the length of the second slice is less than 8? Does core have any function to cut the tail of a slice buffer in a sophisticated way? Or alternatively, if you have reason to believe that the compressed slice buffer will always contain exactly 2 slices and the second slice will always be longer than 8, we can put the reason as comment here and move on.

Reply:
The gzip footer is of 8 bytes so the last slice should have atleast 8 bytes, but I changed the code

Reply:
Cool thanks